src/parquet/arrow/reader.cc

```diff
@@ -44,6 +45,15 @@ using ParquetReader = parquet::ParquetFileReader;
 namespace parquet {
 namespace arrow {
 
+constexpr int64_t kJulianToUnixEpochDays = 2440588l;
+constexpr int64_t kNanosecondsInADay = 86400l * 1000l * 1000l * 1000l;
+
+int64_t impala_timestamp_to_nanoseconds(const Int96& impala_timestamp) {
+  int64_t days_since_epoch = impala_timestamp.value[2] - kJulianToUnixEpochDays;
+  int64_t nanoseconds = *(reinterpret_cast<const int64_t*>(&(impala_timestamp.value)));
+  return days_since_epoch * kNanosecondsInADay + nanoseconds;
```

Comment:
Although there is a lot of code around on various sources on how to convert Julian dates to datetime objects, this is not so complicated as only have to deal with integer Julian dates. That's why I also verified with Spark INT96 output locally.